### ISSUE TITLE
Allow custom Babel config

### ISSUE THREAD

**Issue by lukeed** [cramforce/splittable#36]
In [stage 1]() of the `babelify` transforms, we should be able to inject custom plugins. (and perhaps presets?).

For instance, decorators, async/await, and other ES7 goodies are untouchable right now. Doing this lets Babel compile is down to ES5 or ES6 before GCC sees it.

Some very common babel plugins that were missing & broke my build:

* `babel-plugin-transform-class-properties`
* `babel-plugin-transform-decorators-legacy`
* `babel-preset-stage-0`

The simplest of errors resulted in an unsupported spread operator:

```
ERROR - Parse error. '}' expected
			todo => ({ ...todo, completed })
			           ^
```

**Comment by cramforce** [comment]
Sounds reasonable. A thing is that you almost certainly do not want `babel-preset-stage-0`. For good results it is very important the CC transpiles all the things it can transpile itself.

Any chance you have a sample project handy that could be included into the test suite?

**Comment by lukeed** [comment]
Yeah, likely no presets.

I was converting [Preact's TodoMVC](https://github.com/developit/preact-todomvc).

**Comment by cramforce** [comment]
Cool. I will take a look tonight. Did you have to make any changes to the repo or just run splittable on some input?

**Comment by lukeed** [comment]
Cool, thanks. I just ran Splittable. Normally I spin up Splittable without any issues. But this repo uses more ES6/7 features than I traditionally do.

**Comment by cramforce** [comment]
I have this working

```js
splittable({
  modules: [
    './src/index',
  ],
  writeTo: './sample/out/',
  babel: {
    plugins: [
      'transform-object-rest-spread',
      ['transform-react-jsx', { pragma:'h' }]
    ],
  }
})
```

**Comment by cramforce** [comment]
If I'm not mistaken it is half the size of the rollup output.

**Comment by lukeed** [comment]
Is that doable right now? Or did it rely on changes you made to Splittable locally?

**Comment by cramforce** [comment]
I made local changes. But they seem to work. Will need some time to clean up and write tests.

**Comment by lukeed** [comment]
Cool. Thanks @cramforce !

**Comment by cramforce** [comment]
PR in #41

Needs some final cleanup.

**Comment by lukeed** [comment]
@cramforce Thanks!

**Comment by cramforce** [comment]
@lukeed Please give it a shot. Published as `4.0.0`
